packages/edit-site/src/components/header-edit-mode/index.js

```diff
@@ -258,37 +259,47 @@ export default function HeaderEditMode() {
 
 			<div className="edit-site-header-edit-mode__end">
 				<div className="edit-site-header-edit-mode__actions">
-					{ ! isFocusMode && hasDefaultEditorCanvasView && (
-						<div
-							className={ classnames(
-								'edit-site-header-edit-mode__preview-options',
-								{ 'is-zoomed-out': isZoomedOutView }
-							) }
-						>
-							<PreviewOptions
-								deviceType={ deviceType }
-								setDeviceType={ setPreviewDeviceType }
-								/* translators: button label text should, if possible, be under 16 characters. */
-								viewLabel={ __( 'View' ) }
+					{ ! isFocusMode &&
+						/**
+						 * This isPreviewingTheme() condition can be removed
+						 * once the issues below are fixed
+						 * and previewing the front of the site works well.
+						 *
+						 * https://github.com/WordPress/gutenberg/issues/50713
+						 * https://github.com/WordPress/gutenberg/issues/50712
+						 */
+						! isPreviewingTheme() &&
+						hasDefaultEditorCanvasView && (
+							<div
+								className={ classnames(
+									'edit-site-header-edit-mode__preview-options',
+									{ 'is-zoomed-out': isZoomedOutView }
+								) }
 							>
-								<MenuGroup>
-									<MenuItem
-										href={ homeUrl }
-										target="_blank"
-										icon={ external }
-									>
-										{ __( 'View site' ) }
-										<VisuallyHidden as="span">
-											{
-												/* translators: accessibility text */
-												__( '(opens in a new tab)' )
-											}
-										</VisuallyHidden>
-									</MenuItem>
-								</MenuGroup>
-							</PreviewOptions>
-						</div>
-					) }
+								<PreviewOptions
+									deviceType={ deviceType }
+									setDeviceType={ setPreviewDeviceType }
+									/* translators: button label text should, if possible, be under 16 characters. */
+									viewLabel={ __( 'View' ) }
+								>
+									<MenuGroup>
+										<MenuItem
+											href={ homeUrl }
```

Comment:
I think adding the `isPreviewingTheme()` check here and appending the `?theme_preview` URL query would work well to resolve https://github.com/WordPress/gutenberg/issues/50712. I think https://github.com/WordPress/gutenberg/issues/50713 can be addressed later as a separate PR. The View and Previewing the site links are legitimately useful as is, even if the preview link isn't (yet) appended to all links on the site when previewing.